### ISSUE TITLE
System Report and Commodity Sale Check not firing properly.

### ISSUE THREAD

**Issue by Darkcyde13** [cmdrmcdonald/EliteDangerousDataProvider#109]
I've noticed over the last few days that the System Report and Commodity Sale Check scripts are not always working properly.  Occasionally, parts of them don't seem to be firing, at other times they just don't work at all.

In the System Report, the system.state is often wrong all together.  I've just been able to make a note of the system.state being wrong on two systems that I jumped to consecutively.  Here's the journal entries:-

```
{ "timestamp":"2017-01-27T15:19:19Z", "event":"FSDJump", "StarSystem":"LP 711-32", "StarPos":[9.219,-60.906,-38.250], "SystemAllegiance":"Empire", "SystemEconomy":"$economy_Agri;", "SystemEconomy_Localised":"Agriculture", "SystemGovernment":"$government_Patronage;", "SystemGovernment_Localised":"Patronage", "SystemSecurity":"$SYSTEM_SECURITY_high;", "SystemSecurity_Localised":"High Security", "JumpDist":20.827, "FuelUsed":4.488527, "FuelLevel":22.649828, "SystemFaction":"LP 711-32 Citizens' Forum", "FactionState":"Boom" }
{ "timestamp":"2017-01-27T15:21:08Z", "event":"FSDJump", "StarSystem":"BD-15 447", "StarPos":[8.000,-81.250,-40.125], "SystemAllegiance":"Empire", "SystemEconomy":"$economy_Industrial;", "SystemEconomy_Localised":"Industrial", "SystemGovernment":"$government_Corporate;", "SystemGovernment_Localised":"Corporate", "SystemSecurity":"$SYSTEM_SECURITY_high;", "SystemSecurity_Localised":"High Security", "JumpDist":20.466, "FuelUsed":4.367161, "FuelLevel":27.632839, "SystemFaction":"East India Company", "FactionState":"Expansion" }
```
As you can see the first system is in a Boom state and the second is in an Expansion state.  Here is what EDDI reported:-

```
Telemetry obtained. Pro-cyon Falcon is on route to the L P 7 1 1 dash 32 system. You are now in Empire space. Welcome back sir. This is your first visit to this system. This system is an Empire Patronage supporting Over 864 million people, controlled by Li Yong Rui. L P 7 1 1 dash 32 Citizens' Forum is the controlling faction. There are 4 orbital stations and 4 planetary stations in this system.
Telemetry obtained. Pro-cyon Falcon is travelling to the B D minus 15 4 4 7 system. This is your first visit to this system. This is an Empire Corporate of well over 11 million people, aligned to Li Yong Rui. East India Company is the local faction. There are 3 orbital stations and 2 planetary stations in this system. This system is undergoing a boom; there might be good trading profits to be made here.
```
Here, it said nothing about the system state for LP 711-32, and said BD-15 447 was in a Boom.  I have noticed that it says a system is in a Boom regardless of what the actual state is.  If the state is None, then it generally doesn't say anything, which is correct, but I have had it say Boom sometimes when the state is None.

As for the Commodity Sale Check script, it sometimes reports I have cargo that I sold at the last station I visited, other times it doesn't say anything when I have a hold full of commodities to sell that are in demand at the station I just landed at.

I've added a few lines at various places in the script to see if I can determine what's going wrong, and found that {for cargo in ship.cargo: doesn't always get processed.  I'm guessing maybe it thinks ship.cargo is empty, even when it's not?

It does however, seem to process normally and as I would expect, if I click the Test button.  For example: if it fails to report what cargo I can sell correctly on landing at a station, I can click the test button and it will then work fine.  It seems to work ok after that, but I need to run more tests to be sure.

Would these problems have anything to do with the way EDDI gets it's data, or how the companion app supplies data?

**Comment by cmdrmcdonald** [comment]
At the time the system report is triggered the current state of the system is not available so the scrip falls back to the last known state.  This isn't ideal, and to avoid the general confusion that this can cause I've removed the state information from the jumping script.  When the jump completes the state will be up-to-date, at which point the user can query the state with a VoiceAttack command if they desire.

Sometimes the companion API provides out-of-date data.  I'm slowly going through the discrepancies between companion app and journal information, and commodities is on the list, so this should be sorted at some point after the next release.

**Comment by cmdrmcdonald** [comment]
This is fixed in release 2.2.0.
